packages/devtools_app/lib/src/screens/memory/memory_controller.dart

```diff
@@ -887,6 +849,14 @@ class MemoryController extends DisposableController
         _memoryTracker = null;
       },
     );
+
+    bool _isAndroidChartVisible() =>
+        preferences.memory.androidCollectionEnabled.value &&
+        isConnectedDeviceAndroid;
+    isAndroidChartVisibleNotifier.value = _isAndroidChartVisible();
+    preferences.memory.androidCollectionEnabled.addListener(
```

Comment:
we may want to use `autoDisposeStreamSubscription` so this automatically gets disposed when the memory controller is disposed